### PR TITLE
Add setStatus operation for spanprocessor

### PR DESCRIPTION
**Pull request by TomRoSystems**
**Description:** 

Adding possibility to change status of a span.

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/4754

**Testing:** Added unit test to check functionality.

**Documentation:** Description in spanprocessor readme was extened along with an example.

### CODE REVIEW

File: processor/spanprocessor/README.md [hunk 1]
```diff
@@ -96,5 +97,18 @@ span/to_attributes:
...
 
+### Set status for span
+
+Example:
+
+```yaml
+# Set status allows to set specific status for a given span.
+span/set_status:
+  status:
+    code: 2
```
Comment (tigrannajaryan) [inline processor/spanprocessor/README.md]: It would be nicer to use symbolic status code names (Unset, Ok, Error) instead of numeric codes.

**Review by tigrannajaryan — COMMENTED** [review]
Conceptually looks good to me.

You can also borrow the matching conditions from filterprocessor have have a `apply_to` subsection where you specify which spans to apply the changes to. filterprocessor currently only supports conditions based on attributes, not sure how useful it is as a condition for setting the status.

Also keep in mind there is an initiative to overhaul the processors: https://docs.google.com/document/d/1UNzMG1LBUWpiMiFn1PX4OPVVQsk6EneISbtR4cdBm4Y/edit#

However it is not approved and it is not clear if it will be approved.

File: processor/spanprocessor/README.md (hunk 1 above)
Comment (TomRoSystems) [reply processor/spanprocessor/README.md]: As suggested - please take a look.

File: processor/spanprocessor/factory.go [hunk 2]
```diff
@@ -35,7 +35,8 @@ var processorCapabilities = consumer.Capabilities{MutatesData: true}
 // is not specified.
 // TODO https://github.com/open-telemetry/opentelemetry-collector/issues/215
 //	Move this to the error package that allows for span name and field to be specified.
-var errMissingRequiredField = errors.New("error creating \"span\" processor: either \"from_attributes\" or \"to_attributes\" must be specified in \"name:\"")
+var errMissingRequiredField = errors.New("error creating \"span\" processor: either \"from_attributes\" or \"to_attributes\" must be specified in \"name:\" or \"setStatus\" must be specified")
```
Comment (tigrannajaryan) [inline processor/spanprocessor/factory.go]: Nit: using back quote strings may make this more readable.

File: processor/spanprocessor/factory.go [hunk 3]
```diff
@@ -62,10 +63,16 @@ func createTracesProcessor(
...
 	oCfg := cfg.(*Config)
 	if len(oCfg.Rename.FromAttributes) == 0 &&
-		(oCfg.Rename.ToAttributes == nil || len(oCfg.Rename.ToAttributes.Rules) == 0) {
+		(oCfg.Rename.ToAttributes == nil || len(oCfg.Rename.ToAttributes.Rules) == 0) &&
+		oCfg.SetStatus == nil {
 		return nil, errMissingRequiredField
 	}
 
+	if oCfg.SetStatus != nil &&
+		(oCfg.SetStatus.Code != "Ok" && oCfg.SetStatus.Code != "Err" && oCfg.SetStatus.Code != "Unset") {
```
Comment (tigrannajaryan) [inline processor/spanprocessor/factory.go]: Nit: would be nice to define "Ok", "Err", "Unsert" as consts and use everywhere.

File: processor/spanprocessor/README.md [hunk 4]
```diff
@@ -96,5 +97,21 @@ span/to_attributes:
...
+Example:
+
+```yaml
+# Set status allows to set specific status for a given span. Possible values are
+# Ok, Error and Unset as per
+# https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#set-status
+# The description field allows to set a human-readable message for errors.
+span/set_status:
+  status:
+    code: Error
```
Comment (tigrannajaryan) [inline processor/spanprocessor/README.md:118]: Is it Err or Error? Make sure to match Otel spec.

**Review by tigrannajaryan — COMMENTED** [review]
LGTM, some minor comments.

**Pull request closed without merging**